src/cutadapt/adapters.py

```diff
@@ -890,6 +965,14 @@ def _aligner(self):
         else:
             return self._make_aligner(Where.PREFIX.value)
 
+    def _kmer_finder(self):
+        if isinstance(self.aligner, PrefixComparer):
```

Comment:
The `isinstance()` check makes we wonder whether there’s a better way to structure this.

This should probably be done at a later stage because some other changes are required, but I think one could argue that the k-mer heuristic should be applied by the `Aligner` itself. Then the code in `adapters.py` would not have to be concerned with this. (But one can say the same about the `Prefix-`/`SuffixComparer`, which is why I think this should be done later.)